crosscluster/physical: TestReplicationJobWithReaderTenant failed

crosscluster/physical.TestReplicationJobWithReaderTenant [failed](https://mesolite.cluster.engflow.com/invocations/default/9f5a4e43-572e-4f0c-8171-fdec9a80613c?testReportRun=1&testReportShard=14&testReportAttempt=1#targets-Ly9wa2cvY3Jvc3NjbHVzdGVyL3BoeXNpY2FsOnBoeXNpY2FsX3Rlc3Q=) on master @ [a33ce54e21590e8e94f2ec95f3bff4136c800286](https://github.com/cockroachdb/cockroach/commits/a33ce54e21590e8e94f2ec95f3bff4136c800286):

Fatal error:

```
panic: runtime error: invalid memory address or nil pointer dereference [recovered, repanicked]
[signal SIGSEGV: segmentation violation code=0x1 addr=0x60 pc=0x2e7925f]

```
Stack: 

```
goroutine 31867 [running, "cluster":"destination-readonly"]:
github.com/cockroachdb/cockroach/pkg/sql/colexecerror.CatchVectorizedRuntimeError.func1()
	pkg/sql/colexecerror/error.go:137 +0x546
panic({0x6d314c0?, 0xd27ad00?})
	GOROOT/src/runtime/panic.go:783 +0x132
github.com/cockroachdb/cockroach/pkg/sql/colexecerror.CatchVectorizedRuntimeError.func1()
	pkg/sql/colexecerror/error.go:137 +0x546
panic({0x6d314c0?, 0xd27ad00?})
	GOROOT/src/runtime/panic.go:783 +0x132
github.com/cockroachdb/cockroach/pkg/sql/colexecerror.CatchVectorizedRuntimeError.func1()
	pkg/sql/colexecerror/error.go:137 +0x546
panic({0x6d314c0?, 0xd27ad00?})
	GOROOT/src/runtime/panic.go:783 +0x132
github.com/cockroachdb/cockroach/pkg/sql/row.newTxnKVFetcher.makeSendFunc.makeExternalSpanSendFunc.func3({0x8a13f00, 0xc00c27e960}, 0xc00c8e7540)
	pkg/sql/row/kv_batch_fetcher.go:342 +0x25f
github.com/cockroachdb/cockroach/pkg/sql/row.(*txnKVFetcher).fetch(0xc00caaf860, {0x8a13f00, 0xc00c27e960})
	pkg/sql/row/kv_batch_fetcher.go:659 +0x537
github.com/cockroachdb/cockroach/pkg/sql/row.(*txnKVFetcher).nextBatch(0xc00caaf860, {0x8a13f00, 0xc00c27e960})
	pkg/sql/row/kv_batch_fetcher.go:947 +0x1226
github.com/cockroachdb/cockroach/pkg/sql/row.(*txnKVFetcher).NextBatch(0xc00caaf860, {0x8a13f00?, 0xc00c27e960?})
	pkg/sql/row/kv_batch_fetcher.go:814 +0x98
github.com/cockroachdb/cockroach/pkg/sql/row.(*KVFetcher).nextKV(0xc003116e10, {0x8a13f00, 0xc00c27e960}, 0x0)
	pkg/sql/row/kv_fetcher.go:321 +0xe2
github.com/cockroachdb/cockroach/pkg/sql/row.(*KVFetcher).NextKV(0xc004d40c60?, {0x8a13f00?, 0xc00c27e960?}, 0x2800?)
	pkg/sql/row/kv_fetcher.go:342 +0x4b
github.com/cockroachdb/cockroach/pkg/sql/colfetcher.(*cFetcher).NextBatch(0xc00c0f2d88, {0x8a13f00, 0xc00c27e960})
	pkg/sql/colfetcher/cfetcher.go:776 +0x10d
github.com/cockroachdb/cockroach/pkg/sql/colfetcher.(*ColBatchScan).Next(0xc001bd5a00)
	pkg/sql/colfetcher/colbatch_scan.go:240 +0x2d
github.com/cockroachdb/cockroach/pkg/sql/colexec.(*invariantsChecker).Next(0xc006ae5700)
	pkg/sql/colexec/invariants_checker.go:88 +0x42
github.com/cockroachdb/cockroach/pkg/sql/colexec/colexecutils.(*CancelChecker).Next(0xc006ae5740)
	pkg/sql/colexec/colexecutils/cancel_checker.go:54 +0x2a
github.com/cockroachdb/cockroach/pkg/sql/colexec/colexecbase.(*simpleProjectOp).Next(0xc004bb8f60)
	pkg/sql/colexec/colexecbase/simple_project.go:113 +0x25
github.com/cockroachdb/cockroach/pkg/sql/colexec.(*invariantsChecker).Next(0xc006ae5780)
	pkg/sql/colexec/invariants_checker.go:88 +0x42
github.com/cockroachdb/cockroach/pkg/sql/colflow.(*batchInfoCollector).next(...)
	pkg/sql/colflow/stats.go:120
github.com/cockroachdb/cockroach/pkg/sql/colexecerror.CatchVectorizedRuntimeError(0xc00b1d0060?)
	pkg/sql/colexecerror/error.go:162 +0x4d
github.com/cockroachdb/cockroach/pkg/sql/colflow.(*batchInfoCollector).Next(0xc00bcd4d80)
	pkg/sql/colflow/stats.go:128 +0x75
github.com/cockroachdb/cockroach/pkg/sql/colexec.(*Materializer).next(0xc0028e04e0)
	pkg/sql/colexec/materializer.go:246 +0x67
github.com/cockroachdb/cockroach/pkg/sql/colexec.(*Materializer).nextAdapter(...)
	pkg/sql/colexec/materializer.go:271
github.com/cockroachdb/cockroach/pkg/sql/colexecerror.CatchVectorizedRuntimeError(0x8a13f00?)
	pkg/sql/colexecerror/error.go:162 +0x4d
github.com/cockroachdb/cockroach/pkg/sql/colexec.(*Materializer).Next(0xc0028e04e0)
	pkg/sql/colexec/materializer.go:277 +0x45
github.com/cockroachdb/cockroach/pkg/sql/colflow.(*FlowCoordinator).next(0xc004e76870)
	pkg/sql/colflow/flow_coordinator.go:119 +0x32
github.com/cockroachdb/cockroach/pkg/sql/colflow.(*FlowCoordinator).nextAdapter(...)
	pkg/sql/colflow/flow_coordinator.go:136
github.com/cockroachdb/cockroach/pkg/sql/colexecerror.CatchVectorizedRuntimeError(0xc002adf270?)
	pkg/sql/colexecerror/error.go:162 +0x4d
github.com/cockroachdb/cockroach/pkg/sql/colflow.(*FlowCoordinator).Next(0xc004e76870)
	pkg/sql/colflow/flow_coordinator.go:141 +0x32
github.com/cockroachdb/cockroach/pkg/sql/execinfra.Run({0x8a13f00, 0xc00c27e820}, {0x8a42300, 0xc004e76870}, {0x89ccb70, 0xc00c8a4488})
	pkg/sql/execinfra/base.go:193 +0x45
github.com/cockroachdb/cockroach/pkg/sql/execinfra.(*ProcessorBaseNoHelper).Run(0xc004e76870, {0x8a11ce0?, 0xc00aad5ef0?}, {0x89ccb70, 0xc00c8a4488})
	pkg/sql/execinfra/processorsbase.go:766 +0x66
github.com/cockroachdb/cockroach/pkg/sql/flowinfra.(*FlowBase).Run(0xc00c7f1688, {0x8a11ce0, 0xc00aad5ef0}, 0x80?)
	pkg/sql/flowinfra/flow.go:574 +0x21b
github.com/cockroachdb/cockroach/pkg/sql/colflow.(*vectorizedFlow).Run(0x8aa40e0?, {0x8a11ce0?, 0xc00aad5ef0?}, 0x60?)
	pkg/sql/colflow/vectorized_flow.go:302 +0x213
github.com/cockroachdb/cockroach/pkg/sql.(*DistSQLPlanner).Run(0xc00a720000, {0x8a11ce0, 0xc00aad5ef0}, 0xc0090b51e0, 0xc004a75760, 0xc00f374280, 0xc00c8a4488, 0xc00e68dcd0, 0xc002adfc70)
	pkg/sql/distsql_running.go:1067 +0xedc
github.com/cockroachdb/cockroach/pkg/sql.(*DistSQLPlanner).PlanAndRun(0xc00a720000, {0x8a13f00, 0xc00c0fc8c0}, 0xc00e68dcd0, 0xc0090b51e0, 0xc004a75760, {{0x8a55fe0, 0xc004bb8540}, 0x0}, 0xc00c8a4488, ...)
	pkg/sql/distsql_running.go:2333 +0x245
github.com/cockroachdb/cockroach/pkg/sql.(*DistSQLPlanner).PlanAndRunAll.func3(0x0?, 0x0?, {0x8a13f00?, 0xc00c0fc8c0?}, 0xc002adfd18?, 0x47a794?, 0x100000000?)
	pkg/sql/distsql_running.go:2032 +0xaf
github.com/cockroachdb/cockroach/pkg/sql.(*DistSQLPlanner).PlanAndRunAll(0xc00a720000, {0x8a13f00, 0xc00c0fc8c0}, 0xc00e68dcd0, 0xc0090b51e0, 0xc00e68d710, 0xc00c8a4488, 0x0)
	pkg/sql/distsql_running.go:2035 +0x205
github.com/cockroachdb/cockroach/pkg/sql.(*connExecutor).execWithDistSQLEngine(0xc00e68cf08, {0x8a13f00, 0xc00c0fc8c0}, 0xc00e68d710, 0x7a3f5a1?, {0x8aa6bd0, 0xc00aad4a00}, 0x0, 0xc0034cdd68, {0x0, ...})
	pkg/sql/conn_executor_exec.go:3423 +0x486
github.com/cockroachdb/cockroach/pkg/sql.(*connExecutor).dispatchToExecutionEngine(0xc00e68cf08, {0x8a13f00, 0xc00c0fc8c0}, 0xc00e68d710, {0x8aa6bd0, 0xc00aad4a00})
	pkg/sql/conn_executor_exec.go:2927 +0x108f
github.com/cockroachdb/cockroach/pkg/sql.(*connExecutor).execStmtInOpenState(0xc00e68cf08, {0x8a11ce0?, 0xc00a8749b0?}, {{0x8a3d200, 0xc00a874910}, {0x0, 0x0, 0x0}, {0xc002640cc0, 0x5a}, ...}, ...)
	pkg/sql/conn_executor_exec.go:1082 +0x39ac
github.com/cockroachdb/cockroach/pkg/sql.(*connExecutor).execStmt.func2({0x8a11ce0?, 0xc00a8749b0?})
	pkg/sql/conn_executor_exec.go:176 +0xaa
github.com/cockroachdb/cockroach/pkg/sql.(*connExecutor).execWithProfiling(0x8a11ce0?, {0x8a11ce0?, 0xc00a8749b0?}, {0x8a3d200?, 0xc00a874910?}, 0xc002ae1408?, 0x0?)
	pkg/sql/conn_executor_exec.go:4461 +0x31a
github.com/cockroachdb/cockroach/pkg/sql.(*connExecutor).execStmt(0xc00e68cf08, {0x8a11ce0, 0xc00a8749b0}, {{0x8a3d200, 0xc00a874910}, {0x0, 0x0, 0x0}, {0xc002640cc0, 0x5a}, ...}, ...)
	pkg/sql/conn_executor_exec.go:175 +0x805
github.com/cockroachdb/cockroach/pkg/sql.(*connExecutor).execPortal(0xc00e68cf08, {0x8a11ce0, 0xc00a8749b0}, {{0x0, 0x0}, 0xc00c8df2c0, {0xc001bd5010, 0x1, 0x1}, {0x0, ...}, ...}, ...)
	pkg/sql/conn_executor_exec.go:326 +0x479
github.com/cockroachdb/cockroach/pkg/sql.(*connExecutor).execCmd.func2(0xc002ae1b98, 0xc00e68cf08, 0xc002ae1c90, 0xc002ae1c80, 0xc002ae1c08, 0xc002ae1aa8, 0xc002ae1cc0, 0xc002ae1ca8)
	pkg/sql/conn_executor.go:2538 +0xabc
github.com/cockroachdb/cockroach/pkg/sql.(*connExecutor).execCmd(0xc00e68cf08)
	pkg/sql/conn_executor.go:2540 +0x80a
github.com/cockroachdb/cockroach/pkg/sql.(*connExecutor).run(0xc00e68cf08, {0x8a13f00, 0xc00c08e320}, 0xc005499380?, 0x29c8520?, 0xc00b6cfe30?)
	pkg/sql/conn_executor.go:2339 +0x1f7
github.com/cockroachdb/cockroach/pkg/sql.(*InternalExecutor).runWithEx.func2()
	pkg/sql/internal.go:252 +0x13a
created by github.com/cockroachdb/cockroach/pkg/sql.(*InternalExecutor).runWithEx in goroutine 31822
	pkg/sql/internal.go:249 +0x2b6
```
<details><summary>Log preceding fatal error</summary>
<p>

```
=== RUN   TestReplicationJobWithReaderTenant
    test_log_scope.go:171: test logs captured to: outputs.zip/logTestReplicationJobWithReaderTenant3053608088
    test_log_scope.go:82: use -show-logs to present logs inline
    testutils.go:631: waited for 1.406595455s to advance to 1764868354.181429996,0
```
</p>
</details>

Parameters:
 - <code>attempt=1</code>
 - <code>run=1</code>
 - <code>shard=14</code>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
/cc @cockroachdb/disaster-recovery
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestReplicationJobWithReaderTenant.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>


Jira issue: CRDB-57491

crosscluster/physical.TestReplicationJobWithReaderTenant [failed](https://mesolite.cluster.engflow.com/invocations/default/1b13a05d-2293-4200-8fbc-8b552c29a01e?testReportRun=3&testReportShard=14&testReportAttempt=1#targets-Ly9wa2cvY3Jvc3NjbHVzdGVyL3BoeXNpY2FsOnBoeXNpY2FsX3Rlc3Q=) on master @ [41bf2cb26c0a91b457cc8c4d93ba87cb4f771ffe](https://github.com/cockroachdb/cockroach/commits/41bf2cb26c0a91b457cc8c4d93ba87cb4f771ffe):

Fatal error:

```
panic: runtime error: invalid memory address or nil pointer dereference [recovered, repanicked]
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x4cadef9]

```
Stack: 

```
goroutine 33098 [running, "cluster":"destination-readonly"]:
github.com/cockroachdb/cockroach/pkg/sql/colexecerror.CatchVectorizedRuntimeError.func1()
	pkg/sql/colexecerror/error.go:137 +0x81a
panic({0xafea1e0?, 0x1218ed20?})
	GOROOT/src/runtime/panic.go:783 +0x132
github.com/cockroachdb/cockroach/pkg/sql/colexecerror.CatchVectorizedRuntimeError.func1()
	pkg/sql/colexecerror/error.go:137 +0x81a
panic({0xafea1e0?, 0x1218ed20?})
	GOROOT/src/runtime/panic.go:783 +0x132
github.com/cockroachdb/cockroach/pkg/sql/colexecerror.CatchVectorizedRuntimeError.func1()
	pkg/sql/colexecerror/error.go:137 +0x81a
panic({0xafea1e0?, 0x1218ed20?})
	GOROOT/src/runtime/panic.go:783 +0x132
github.com/cockroachdb/cockroach/pkg/sql/row.newTxnKVFetcher.makeSendFunc.makeExternalSpanSendFunc.func3({0xd50a620, 0xc0048ae820}, 0xc0094ae3c0)
	pkg/sql/row/kv_batch_fetcher.go:342 +0x3f9
github.com/cockroachdb/cockroach/pkg/sql/row.(*txnKVFetcher).fetch(0xc00c8ae5a0, {0xd50a620, 0xc0048ae820})
	pkg/sql/row/kv_batch_fetcher.go:659 +0xd15
github.com/cockroachdb/cockroach/pkg/sql/row.(*txnKVFetcher).nextBatch(0xc00c8ae5a0, {0xd50a620, 0xc0048ae820})
	pkg/sql/row/kv_batch_fetcher.go:947 +0x1f0a
github.com/cockroachdb/cockroach/pkg/sql/row.(*txnKVFetcher).NextBatch(0xc00c8ae5a0, {0xd50a620, 0xc0048ae820})
	pkg/sql/row/kv_batch_fetcher.go:814 +0xb8
github.com/cockroachdb/cockroach/pkg/sql/row.(*KVFetcher).nextKV(0xc002cb6ab0, {0xd50a620, 0xc0048ae820}, 0x0)
	pkg/sql/row/kv_fetcher.go:321 +0x162
github.com/cockroachdb/cockroach/pkg/sql/row.(*KVFetcher).NextKV(0xc002cb6ab0, {0xd50a620, 0xc0048ae820}, 0x0)
	pkg/sql/row/kv_fetcher.go:342 +0x90
github.com/cockroachdb/cockroach/pkg/sql/colfetcher.(*cFetcher).NextBatch(0xc00bd41b08, {0xd50a620, 0xc0048ae820})
	pkg/sql/colfetcher/cfetcher.go:776 +0x1ec
github.com/cockroachdb/cockroach/pkg/sql/colfetcher.(*ColBatchScan).Next(0xc001f83190)
	pkg/sql/colfetcher/colbatch_scan.go:240 +0x94
github.com/cockroachdb/cockroach/pkg/sql/colexec.(*invariantsChecker).Next(0xc00358c8c0)
	pkg/sql/colexec/invariants_checker.go:88 +0x6c
github.com/cockroachdb/cockroach/pkg/sql/colexec/colexecutils.(*CancelChecker).Next(0xc00358c940)
	pkg/sql/colexec/colexecutils/cancel_checker.go:54 +0x45
github.com/cockroachdb/cockroach/pkg/sql/colexec.(*invariantsChecker).Next(0xc00358c980)
	pkg/sql/colexec/invariants_checker.go:88 +0x6c
github.com/cockroachdb/cockroach/pkg/sql/colexec.(*allSpooler).spool(0xc009552420)
	pkg/sql/colexec/sort.go:132 +0x8b
github.com/cockroachdb/cockroach/pkg/sql/colexec.(*sortOp).Next(0xc002db4580)
	pkg/sql/colexec/sort.go:285 +0xbb
github.com/cockroachdb/cockroach/pkg/sql/colexec/colexecdisk.(*diskSpillerBase).Next.func1()
	pkg/sql/colexec/colexecdisk/disk_spiller.go:246 +0x51
github.com/cockroachdb/cockroach/pkg/sql/colexecerror.CatchVectorizedRuntimeError(0xc003c09a90)
	pkg/sql/colexecerror/error.go:162 +0x79
github.com/cockroachdb/cockroach/pkg/sql/colexec/colexecdisk.(*diskSpillerBase).Next(0xc004712d80)
	pkg/sql/colexec/colexecdisk/disk_spiller.go:244 +0xe5
github.com/cockroachdb/cockroach/pkg/sql/colexec/colexecbase.(*simpleProjectOp).Next(0xc0095524e0)
	pkg/sql/colexec/colexecbase/simple_project.go:113 +0x50
github.com/cockroachdb/cockroach/pkg/sql/colexec.(*invariantsChecker).Next(0xc00358ca80)
	pkg/sql/colexec/invariants_checker.go:88 +0x6c
github.com/cockroachdb/cockroach/pkg/sql/colexec.(*Materializer).next(0xc003f0b520)
	pkg/sql/colexec/materializer.go:246 +0xf5
github.com/cockroachdb/cockroach/pkg/sql/colexec.(*Materializer).nextAdapter(...)
	pkg/sql/colexec/materializer.go:271
github.com/cockroachdb/cockroach/pkg/sql/colexecerror.CatchVectorizedRuntimeError(0xc003c09d80)
	pkg/sql/colexecerror/error.go:162 +0x79
github.com/cockroachdb/cockroach/pkg/sql/colexec.(*Materializer).Next(0xc003f0b520)
	pkg/sql/colexec/materializer.go:277 +0x85
github.com/cockroachdb/cockroach/pkg/sql/colflow.(*FlowCoordinator).next(0xc0061c0a50)
	pkg/sql/colflow/flow_coordinator.go:119 +0x66
github.com/cockroachdb/cockroach/pkg/sql/colflow.(*FlowCoordinator).nextAdapter(...)
	pkg/sql/colflow/flow_coordinator.go:136
github.com/cockroachdb/cockroach/pkg/sql/colexecerror.CatchVectorizedRuntimeError(0xc003c09ef8)
	pkg/sql/colexecerror/error.go:162 +0x79
github.com/cockroachdb/cockroach/pkg/sql/colflow.(*FlowCoordinator).Next(0xc0061c0a50)
	pkg/sql/colflow/flow_coordinator.go:141 +0x5f
github.com/cockroachdb/cockroach/pkg/sql/execinfra.Run({0xd50a620, 0xc0048ae6e0}, {0xd539080, 0xc0061c0a50}, {0xd4c0da0, 0xc00bd40488})
	pkg/sql/execinfra/base.go:193 +0x4a
github.com/cockroachdb/cockroach/pkg/sql/execinfra.(*ProcessorBaseNoHelper).Run(0xc0061c0a50, {0xd508400, 0xc00d8c5b30}, {0xd4c0da0, 0xc00bd40488})
	pkg/sql/execinfra/processorsbase.go:766 +0xcd
github.com/cockroachdb/cockroach/pkg/sql/flowinfra.(*FlowBase).Run(0xc00bd44008, {0xd508400, 0xc00d8c5b30}, 0x0)
	pkg/sql/flowinfra/flow.go:574 +0x77b
github.com/cockroachdb/cockroach/pkg/sql/colflow.(*vectorizedFlow).Run(0xc00d8c5ae0, {0xd508400, 0xc00d8c5b30}, 0x0)
	pkg/sql/colflow/vectorized_flow.go:302 +0x38e
github.com/cockroachdb/cockroach/pkg/sql.(*DistSQLPlanner).Run(0xc00c8f5340, {0xd508400, 0xc00d8c5b30}, 0xc00c7089c0, 0xc003f1ab00, 0xc00d1b4780, 0xc00bd40488, 0xc00b7552d8, 0xc003c0ac50)
	pkg/sql/distsql_running.go:1067 +0x1b96
github.com/cockroachdb/cockroach/pkg/sql.(*DistSQLPlanner).PlanAndRun(0xc00c8f5340, {0xd508400, 0xc00d8c5860}, 0xc00b7552d8, 0xc00c7089c0, 0xc003f1ab00, {{0xd54c680, 0xc00833fda0}, 0x0}, 0xc00bd40488, ...)
	pkg/sql/distsql_running.go:2333 +0x365
github.com/cockroachdb/cockroach/pkg/sql.(*DistSQLPlanner).PlanAndRunAll.func3(0xc00b754d18, 0xc00c8f5340, {0xd508400, 0xc00d8c5860}, 0xc00b7552d8, 0xc00c7089c0, 0xc00bd40488)
	pkg/sql/distsql_running.go:2032 +0x19d
github.com/cockroachdb/cockroach/pkg/sql.(*DistSQLPlanner).PlanAndRunAll(0xc00c8f5340, {0xd508400, 0xc00d8c5860}, 0xc00b7552d8, 0xc00c7089c0, 0xc00b754d18, 0xc00bd40488, 0x0)
	pkg/sql/distsql_running.go:2035 +0x359
github.com/cockroachdb/cockroach/pkg/sql.(*connExecutor).execWithDistSQLEngine(0xc00b754508, {0xd508400, 0xc00d8c5860}, 0xc00b754d18, 0x3, {0xd5a8410, 0xc00d8c5810}, 0x1, 0xc003f1ac38, {0x0, ...})
	pkg/sql/conn_executor_exec.go:3423 +0xbc5
github.com/cockroachdb/cockroach/pkg/sql.(*connExecutor).dispatchToExecutionEngine(0xc00b754508, {0xd508400, 0xc00d8c5860}, 0xc00b754d18, {0xd5a8410, 0xc00d8c5810})
	pkg/sql/conn_executor_exec.go:2927 +0x21b5
github.com/cockroachdb/cockroach/pkg/sql.(*connExecutor).execStmtInOpenState(0xc00b754508, {0xd50a620, 0xc004f7de00}, {{0xd533e80, 0xc00d8c4ff0}, {0x0, 0x0, 0x0}, {0xbfaa1a2, 0x125}, ...}, ...)
	pkg/sql/conn_executor_exec.go:1082 +0x6070
github.com/cockroachdb/cockroach/pkg/sql.(*connExecutor).execStmt.func2({0xd50a620, 0xc004f7de00})
	pkg/sql/conn_executor_exec.go:176 +0x185
github.com/cockroachdb/cockroach/pkg/sql.(*connExecutor).execWithProfiling(0xc00b754508, {0xd50a620, 0xc004f7de00}, {0xd533e80, 0xc00d8c4ff0}, 0xc00dfe3680, 0xc003c0cf30)
	pkg/sql/conn_executor_exec.go:4461 +0x577
github.com/cockroachdb/cockroach/pkg/sql.(*connExecutor).execStmt(0xc00b754508, {0xd50a620, 0xc004f7de00}, {{0xd533e80, 0xc00d8c4ff0}, {0x0, 0x0, 0x0}, {0xbfaa1a2, 0x125}, ...}, ...)
	pkg/sql/conn_executor_exec.go:175 +0xdbe
github.com/cockroachdb/cockroach/pkg/sql.(*connExecutor).execPortal(0xc00b754508, {0xd50a620, 0xc004f7de00}, {{0x0, 0x0}, 0xc00dfe3680, {0xc001f82bf0, 0x1, 0x1}, {0x0, ...}, ...}, ...)
	pkg/sql/conn_executor_exec.go:326 +0x73f
github.com/cockroachdb/cockroach/pkg/sql.(*connExecutor).execCmd.func2(0xc003c0d9b8, 0xc00b754508, 0xc003c0dae8, 0xc003c0dad8, 0xc003c0dac0, 0xc003c0d788, 0xc003c0db38, 0xc003c0db08)
	pkg/sql/conn_executor.go:2538 +0x13a9
github.com/cockroachdb/cockroach/pkg/sql.(*connExecutor).execCmd(0xc00b754508)
	pkg/sql/conn_executor.go:2540 +0x115d
github.com/cockroachdb/cockroach/pkg/sql.(*connExecutor).run(0xc00b754508, {0xd50a620, 0xc004f7d540}, 0xc0059a38c0, 0xc00a352288, 0x0)
	pkg/sql/conn_executor.go:2339 +0x409
github.com/cockroachdb/cockroach/pkg/sql.(*InternalExecutor).runWithEx.func2()
	pkg/sql/internal.go:252 +0x285
created by github.com/cockroachdb/cockroach/pkg/sql.(*InternalExecutor).runWithEx in goroutine 33079
	pkg/sql/internal.go:249 +0x55f
```
<details><summary>Log preceding fatal error</summary>
<p>

```
=== RUN   TestReplicationJobWithReaderTenant
    test_log_scope.go:171: test logs captured to: outputs.zip/logTestReplicationJobWithReaderTenant3967488258
    test_log_scope.go:82: use -show-logs to present logs inline
    testutils.go:631: waited for 3.664929194s to advance to 1765023627.900004402,0
```
</p>
</details>

Parameters:
 - <code>attempt=1</code>
 - <code>race=true</code>
 - <code>run=3</code>
 - <code>shard=14</code>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestReplicationJobWithReaderTenant.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>

--- a/pkg/sql/row/kv_batch_fetcher.go
+++ b/pkg/sql/row/kv_batch_fetcher.go
@@ -338,15 +338,18 @@ func makeExternalSpanSendFunc(
 				}
 				return nil
 			})
-
-		if res.CPUTime > 0 {
-			atomic.AddInt64(kvCPUTime, res.CPUTime)
-		}
 		// Note that in some code paths there is no concurrency when using the
 		// sendFunc, but we choose to unconditionally use atomics here since its
 		// overhead should be negligible in the grand scheme of things anyway.
 		atomic.AddInt64(batchRequestsIssued, 1)
-		return res, err
+		if err != nil {
+			return nil, err
+		}
+
+		if res.CPUTime > 0 {
+			atomic.AddInt64(kvCPUTime, res.CPUTime)
+		}
+		return res, nil
 	}
 }
 

